### PR TITLE
icu: fix build on mingw@windows

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **icu/67.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]

File: recipes/icu/all/conanfile.py [hunk 1]
```diff
@@ -234,16 +234,28 @@ def _data_filename(self):
...
+
         # icudata
         self.cpp_info.components["icu-data"].names["cmake_find_package"] = "data"
         self.cpp_info.components["icu-data"].names["cmake_find_package_multi"] = "data"
         self.cpp_info.components["icu-data"].libs = [self._lib_name("icudt" if self.settings.os == "Windows" else "icudata")]
         if not self.options.shared:
             self.cpp_info.components["icu-data"].defines.append("U_STATIC_IMPLEMENTATION")
+
+        # icu uses c++, so add the c++ runtime
+        libcxx = self.settings.get_safe("compiler.libcxx")
```
Comment (SSE4) [inline recipes/icu/all/conanfile.py:246]: why not tools.cppstd_library?

Comment (madebr) [reply recipes/icu/all/conanfile.py:246]: I know of [`tools.cppstd_flag`](https://docs.conan.io/en/latest/reference/tools.html#tools-cppstd-flag), but not `tools.cppstd_library`.

File: recipes/icu/all/conanfile.py [hunk 2]
```diff
@@ -234,16 +234,28 @@ def _data_filename(self):
...
+        if libcxx in ["libstdc++", "libstdc++11"]:
+            self.cpp_info.components["icu-data"].system_libs.append("stdc++")
+        elif libcxx == "libc++":
+            self.cpp_info.components["icu-data"].system_libs.append("c++")
+        elif libcxx in ["c++_static", "c++_shared"]:
+            self.cpp_info.components["icu-data"].system_libs.extend([libcxx, "c++abi"])
```
Comment (madebr) [inline recipes/icu/all/conanfile.py:252]:
```suggestion

        if tools.stdcpp_library(self):
            self.cpp_info.components["icu-data"].system_libs.append(tools.stdcpp_library(self)
```

File: recipes/icu/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/icu/all/conanfile.py:246]: Ok, you meant [tools.stdcpp_library](https://docs.conan.io/en/latest/reference/tools.html#tools-stdcpp-library)

File: recipes/icu/all/conanfile.py [hunk 3]
```diff
@@ -234,16 +234,23 @@ def _data_filename(self):
...
         # icudata
         self.cpp_info.components["icu-data"].names["cmake_find_package"] = "data"
         self.cpp_info.components["icu-data"].names["cmake_find_package_multi"] = "data"
         self.cpp_info.components["icu-data"].libs = [self._lib_name("icudt" if self.settings.os == "Windows" else "icudata")]
         if not self.options.shared:
             self.cpp_info.components["icu-data"].defines.append("U_STATIC_IMPLEMENTATION")
+
+        # icu uses c++, so add the c++ runtime
+        if tools.stdcpp_library(self):
+            self.cpp_info.components["icu-data"].system_libs.append(tools.stdcpp_library(self)
```
Comment (madebr) [inline recipes/icu/all/conanfile.py:247]:
```suggestion
            self.cpp_info.components["icu-data"].system_libs.append(tools.stdcpp_library(self))
```

**Review by danimtb — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**